src/components/CodeManagement/index.jsx

```diff
@@ -25,33 +25,25 @@ class CodeManagement extends React.Component {
   }
 
   componentDidMount() {
-    const { enterpriseId } = this.props;
-
+    const { enterpriseId, location } = this.props;
     if (enterpriseId) {
       this.props.fetchCouponOrders();
     }
+    if (location.state && location.state.hasRequestedCodes) {
+      this.setState({ // eslint-disable-line react/no-did-mount-set-state
+        hasRequestedCodes: location.state.hasRequestedCodes,
+      });
+    }
   }
 
   componentDidUpdate(prevProps) {
-    const { enterpriseId, location, history } = this.props;
-
+    const { enterpriseId } = this.props;
     if (enterpriseId && enterpriseId !== prevProps.enterpriseId) {
       this.props.fetchCouponOrders();
     }
-
-    if (location.state && location.state.hasRequestedCodes && location !== prevProps.location) {
-      this.setState({ // eslint-disable-line react/no-did-update-set-state
-        hasRequestedCodes: location.state.hasRequestedCodes,
-      });
-
-      // reset location state so the "requested more codes" status alert disappears
-      history.replace({
```

Comment:
I think this `history.replace({ ... })` part also should be moved up to `componentDidMount`.

If you request more codes, the success status alert appears. If you then go to the request more codes page, and then immediately back to the code management page, the status alert still appears. I would expect it to disappear when you switch pages.